packages/cli/test/unit/commands/build/index.test.ts

```diff
@@ -1250,43 +1250,63 @@ describe('build', () => {
       (await fs.readFile(join(output, 'static/index.txt'), 'utf8')).trim()
     ).toEqual('marketing');
   });
-});
 
-it('should create symlinks for duplicate references to Lambda / EdgeFunction instances', async () => {
-  if (process.platform === 'win32') {
-    console.log('Skipping test on Windows');
-    return;
-  }
-  const cwd = fixture('functions-symlink');
-  const output = join(cwd, '.vercel/output');
-  client.cwd = cwd;
-  const exitCode = await build(client);
-  expect(exitCode).toEqual(0);
-
-  // "functions" directory has output Functions
-  const functions = await fs.readdir(join(output, 'functions'));
-  expect(functions.sort()).toEqual([
-    'edge.func',
-    'edge2.func',
-    'lambda.func',
-    'lambda2.func',
-  ]);
-  expect(
-    fs.lstatSync(join(output, 'functions/lambda.func')).isDirectory()
-  ).toEqual(true);
-  expect(
-    fs.lstatSync(join(output, 'functions/edge.func')).isDirectory()
-  ).toEqual(true);
-  expect(
-    fs.lstatSync(join(output, 'functions/lambda2.func')).isSymbolicLink()
-  ).toEqual(true);
-  expect(
-    fs.lstatSync(join(output, 'functions/edge2.func')).isSymbolicLink()
-  ).toEqual(true);
-  expect(fs.readlinkSync(join(output, 'functions/lambda2.func'))).toEqual(
-    'lambda.func'
-  );
-  expect(fs.readlinkSync(join(output, 'functions/edge2.func'))).toEqual(
-    'edge.func'
-  );
+  it('should write to flags.json', async () => {
+    const cwd = fixture('with-flags');
+    const output = join(cwd, '.vercel', 'output');
+
+    client.cwd = cwd;
+    client.setArgv('build', '--yes');
+
+    const exitCode = await build(client);
+    expect(exitCode).toEqual(0);
+
+    expect(fs.existsSync(join(output, 'flags.json'))).toBe(true);
+    expect(fs.readJSONSync(join(output, 'flags.json'))).toEqual({
+      definitions: {
+        'my-next-flag': {
+          options: [{ value: true }, { value: false }],
+        },
+      },
+    });
+  });
+
+  it('should create symlinks for duplicate references to Lambda / EdgeFunction instances', async () => {
```

Comment:
I moved this test into the `describe` block, since I assume this was the reason it failed sometimes.
Probably because the `beforeEach` in the describe block resets the env variable, which caused this test to appear as if it was running within `vercel build`, causing a failure.